### PR TITLE
Do not log the latest version if it's unknown

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

It is possible when telemetry is enabled, but the report wasn't sent yet.

Additionally, clarify the differences between `undecided` and `enabled`.

## Readiness checklist

* [x] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [x] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/pg/msg_getlog.go [hunk 1]
```diff
@@ -83,27 +83,34 @@ func (h *Handler) MsgGetLog(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
...
 			"Please star us on GitHub: https://github.com/FerretDB/FerretDB.",
 		}
 
-		state := h.StateProvider.Get()
-
 		switch {
 		case state.Telemetry == nil:
 			startupWarnings = append(
 				startupWarnings,
-				"The telemetry state is undecided; the first report will be sent soon. "+
```
Comment (AlekSi) [inline internal/handlers/pg/msg_getlog.go:100]: We should not say "the first report will be sent soon" after the first report is sent.

File: internal/util/state/state.go [hunk 2]
```diff
@@ -32,8 +33,36 @@ type State struct {
...
+func (s *State) UpdateAvailable() bool {
+	// if telemetry was enabled and then disabled, then LatestVersion will never be updated
+	if s.Telemetry != nil && !*s.Telemetry {
+		return false
+	}
+
+	// if we don't know yet
+	if s.LatestVersion == "" {
+		return false
+	}
```
Comment (AlekSi) [inline internal/util/state/state.go:63]: That's what was missing

File: website/docs/telemetry.md [hunk 3]
```diff
@@ -35,23 +35,30 @@ Argument values, data field names, successful responses, or error messages are n
 ## Version notification
 
 When a FerretDB update is available, the telemetry service sends periodic reports containing information about the latest FerretDB version.
-
-This information is logged in the server logs, `startupWarnings`, and `serverStatus` command output.
```
Comment (AlekSi) [inline website/docs/telemetry.md:39]: `serverStatus` never logged version update information, only the current telemetry state

File: website/docs/telemetry.md [hunk 4]
```diff
@@ -35,23 +35,30 @@ Argument values, data field names, successful responses, or error messages are n
...
 
 :::info
-When the state setting is `undecided`, users have a delay period of one hour after startup before telemetry data is reported.
+`undecided` state does not change itself into `enabled` after the first or any other report.
+Only explicit user action (see below) could change it to `enabled` or `disabled`.
 :::
 
 ### Disable telemetry
 
-We urge you not to disable this service, as its insights will help us enhance our software.
```
Comment (AlekSi) [inline website/docs/telemetry.md:54]: Users can't disable _service_ (beacon.ferretdb.io)

**Review by w84thesun — APPROVED** [review]
👍

**Review by noisersup — DISMISSED** [review]

File: website/docs/telemetry.md [hunk 5]
```diff
@@ -35,23 +35,30 @@ Argument values, data field names, successful responses, or error messages are n
...
+* When `enabled`, the first report is sent right after FerretDB starts.
+  If `undecided`, the first report is delayed by one hour.
+  That should give you enough time to disable it if you decide to do so.
+* Similarly, when `enabled`, the last report is sent right before FerretDB shutdowns.
+  That does not happen when `undecided`.
 
 :::info
-When the state setting is `undecided`, users have a delay period of one hour after startup before telemetry data is reported.
+`undecided` state does not change itself into `enabled` after the first or any other report.
+Only explicit user action (see below) could change it to `enabled` or `disabled`.
```
Comment (Fashander) [inline website/docs/telemetry.md:56]:
```suggestion
`undecided` state does not automatically change into `enabled` or `disabled` after the first or any other report.
Explicit user action is required (see below) to change an `undecided` state to `enabled` or `disabled`.
```

File: website/docs/telemetry.md [hunk 6]
```diff
@@ -35,23 +35,30 @@ Argument values, data field names, successful responses, or error messages are n
...
 
 ## Configure telemetry
 
 The telemetry service has three state settings: `enabled`, `disabled`, and `undecided` (default).
+The latter acts as if the telemetry reporter is `enabled` with two differences:
+
+* When `enabled`, the first report is sent right after FerretDB starts.
+  If `undecided`, the first report is delayed by one hour.
+  That should give you enough time to disable it if you decide to do so.
+* Similarly, when `enabled`, the last report is sent right before FerretDB shutdowns.
```
Comment (Fashander) [inline website/docs/telemetry.md:51]:
```suggestion
* Similarly, when `enabled`, the last report is sent right before FerretDB shuts down.
```

**Review by Fashander — CHANGES_REQUESTED** [review]
Looks good. I left a few suggestions on some of the wordings

File: website/docs/telemetry.md [hunk 7]
```diff
@@ -35,23 +35,30 @@ Argument values, data field names, successful responses, or error messages are n
...
-This information is logged in the server logs, `startupWarnings`, and `serverStatus` command output.
+This information is logged in the server logs and `startupWarnings` command output.
 
 While you may not upgrade to the latest release immediately,
 ensure that you update early to take advantage of recent bug fixes, new features, and performance improvements.
 
 ## Configure telemetry
 
 The telemetry service has three state settings: `enabled`, `disabled`, and `undecided` (default).
+The latter acts as if the telemetry reporter is `enabled` with two differences:
```
Comment (rumyantseva) [inline website/docs/telemetry.md:46]: What "the latter" refers to? If it's about `undecided`, let's say `undecided` for clarity.

**Review by rumyantseva — APPROVED** [review]
`:ack:`

File: website/docs/telemetry.md (hunk 7 above)
Comment (AlekSi) [reply website/docs/telemetry.md:46]: I'm not sure if that would improve clarity (what else "the latter" could mean?), but that would definitely increase tautology by using the same word next to each other

**Review by rumyantseva — APPROVED** [review]
`:ack:`

**Review by Fashander — DISMISSED** [review]

**Review by w84thesun — APPROVED** [review]

**Pull request merged**